priam/src/main/java/com/netflix/priam/identity/InstanceIdentity.java

```diff
@@ -267,12 +267,6 @@ private void populateInstancesByAvailabilityZoneMultiMap() {
             seeds.add(instancesByAvailabilityZoneMultiMap.get(loc).get(0).getHostIP());
         }
 
-        // Remove this node from the seed list so Cassandra auto-bootstrap will kick in.  Unless this is the only node
-        // in the cluster.
-        if (seeds.size() > 1) {
-            seeds.remove(myInstance.getHostIP());
-        }
-
```

Comment:
Basically, this is an abuse of a non-feature. We configure Cassandra to have autobootstrap = true all the time. Because of this, if no seeds are online (specifically, if gossip is not running for any of the provided seeds), Cassandra will not start. Previously, it would start. Instead of relying on this, we now consider ourselves a seed if there are no other seeds available. This allows Cassandra to start. Once Cassandra has started, other nodes can gossip with it as seeds.

An alternative implementation, which may be the more correct one, would be to set autobootstrap to false for the first (and only the first) node to attempt to start. That is a difficult synchronization problem, though, and not any more reliable than this method, even if it is more correct.